packages/server/modules/core/domain/objects/types.ts

```diff
@@ -1,3 +1,26 @@
-import { ObjectRecord } from '@/modules/core/helpers/types'
+import { ObjectChildrenClosureRecord, ObjectRecord } from '@/modules/core/helpers/types'
+import { Nullable } from '@speckle/shared'
+import { OverrideProperties, SetOptional } from 'type-fest'
+
+export type SpeckleObjectClosureEntry = ObjectChildrenClosureRecord
 
 export type SpeckleObject = ObjectRecord
+
+/**
+ * We preemptively serialize a couple of fields (usually knex does it)
+ */
+export type InsertableSpeckleObject = OverrideProperties<
+  SetOptional<SpeckleObject, 'createdAt'>,
+  {
+    data: string
+    totalChildrenCountByDepth: string
+  }
+>
+
+export type RawSpeckleObject = Record<string, unknown> & {
```

Comment:
@didimitrie @gjedlicska Does this make sense for the type of the raw object sent into the API (not the final object DB record)? I'm not sure about 'totalChildrenCount' and '__closure' - maybe they should be optional? I was thrown off by a `__closure !== null` check which made me think `undefined` is not a supported value there

Reply:
I think `id` could be optional as well, it gets generated by prepInsertionObject. speckle_type seems to be the only actually required prop

Reply:
See final versions of InsertableSpeckleObject & RawSPeckleObject here - https://github.com/specklesystems/speckle-server/pull/3335/files#diff-d35068824944d012d2120242ee1ed42bfc11ca8d6a257455e3039c48a3f0776d